client-lite/src/download/download_manager.cpp

```diff
@@ -102,12 +102,20 @@ DownloadStatus DownloadManager::GetDownloadStatus(const std::string& downloadId)
 bool DownloadManager::IsIdle() const
 {
     // Reset _fRunning if we are idle to disallow new downloads.
-    // This handles the race between shutdown and new download request coming in.
+    // This handles the race between shutdown or policy refresh and new download request coming in.
     std::unique_lock<std::shared_timed_mutex> lock(_downloadsMtx);
     _fRunning = !_downloads.empty();
     return !_fRunning;
 }
 
+void DownloadManager::RefreshConfigs() const
+{
+    _taskThread.SchedBlock([&]()
```

Comment:
>SchedBlock [](start = 16, length = 10)

No need to block, so use Sched() with 'this' as the tag.

Reply:
Should be SchedImmediate() so downloads immediately get the new configs I think

---
In reply to: [594674119](https://github.com/microsoft/do-client/pull/11#discussion_r594674119) [](ancestors = 594674119)